### PR TITLE
✨ add versioning information to controller

### PR DESCRIPTION
**Pull request by kranurag7**
This commit adds versioning information to the controller when it starts.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -119,6 +120,12 @@ func main() {
 	// Setup the context that's going to be used in controllers and for the manager.
 	ctx := ctrl.SetupSignalHandler()
 
+	setupLog.Info("starting manager", "version", caphversion.Get().String())
+	if err := mgr.Start(ctx); err != nil {
```
Comment (janiskemper) [inline main.go:124]: you have this code already in https://github.com/syself/cluster-api-provider-hetzner/pull/958/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R224 where you can just add the version infos

**Review by janiskemper — CHANGES_REQUESTED** [review]
can we test this with tilt?

File: pkg/version/version.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	"fmt"
+	"runtime"
+
+	"github.com/pkg/errors"
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+const (
```
Comment (batistein) [inline pkg/version/version.go:30]: I think we can remove these const

**Review by batistein — CHANGES_REQUESTED** [review]

File: pkg/version/version.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	"runtime"
+
+	"github.com/pkg/errors"
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+// CheckKubernetesVersion return an error if the Kubernetes version in a cluster is lower than the specified minK8sVersion.
+func CheckKubernetesVersion(config *rest.Config, minK8sVersion string) error {
```
Comment (batistein) [inline pkg/version/version.go:31]: from where does we get minK8sVersion? Is it necessary at all?

**Review by batistein — CHANGES_REQUESTED** [review]

**Review by batistein — COMMENTED** [review]
looks good now

**Comment by janiskemper** [comment]
we should add it here https://github.com/syself/cluster-api-provider-hetzner/pull/692

**Comment by kranurag7** [comment]
> we should add it here https://github.com/syself/cluster-api-provider-hetzner/pull/692

Me and @batistein agreed on doing a separate PR for that purpose, but In the latest commit, I've included that. 

We also agreed to log the version information at the start of the controller.

**Review by janiskemper — APPROVED** [review]
mmh okay. I guess we have so many PRs open that we should be careful with them.. We have a huge queue of e2e tests right now. 
@batistein you say whether we make two commits or one. 
For me
LGTM

**Comment by batistein** [comment]
one is fine!

**Pull request merged**